---
title: "general_figures2"
output: html_document
---

see the comments in general_figures.Rmd for why we even need this. but long story short is to not trust anything in all_data/dPSI25_no_merging_all_reps that came from that R script (GO stuff mostly) cause the AS event filtering was done wrong.

```{r}
library(tidyverse)
library(ggVennDiagram)
library(ggstance)
library(gghighlight)
library(ggrepel)
```

```{r}
inc <- read_delim("C:/Users/hasna/Documents/MSc_project/vasttools/all_data/complete_inclusion_table.tab", delim = "\t")
```

figure 1: AS events between sample groups. right now I'm thinking this would work best as a venn diagram, but if there's not a lot of overlap then maybe I'll convert it to a table. I'm expecting overlaps between the groups that are reasonably similar (ex. watered samples across timepoints, since that'll just be an age effect). To start with, I'm gonna use 25% sequence inclusion / splice site usage as my cutoff to say that AS occurred (I think this makes sense since that's also my dAS cutoff, so a 0 vs 25% sequence inclusion would have been considered an AS event)

1.1: construct a df for each treatment group. This will allow us to filter down to events where all reps have sufficient read coverage, and also average the replicates, and choose the events with AS

```{r}
K20d <- inc %>% select(c("EVENT", contains("K20d"))) %>% 
  mutate("pass" = ifelse(
  str_sub(`D6K20d2_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6K20d3_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K20d2_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K20d5_191011_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
K20d_events <- K20d$EVENT

K40d <- inc %>% select(c("EVENT", contains("K40d"))) %>% 
  mutate("pass" = ifelse(
  str_sub(`D7K40d1_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K40d2_191011_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
K40d_events <- K40d$EVENT

K60d <- inc %>% select(c("EVENT", contains("K60d"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D7K60d1_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K60d2_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K60d3_191011_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
K60d_events <- K60d$EVENT

K20w <- inc %>% select(c("EVENT", contains("K20W"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D7K20W1_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K20W2_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6K20W1_191011_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
K20w_events <- K20w$EVENT

K40w <- inc %>% select(c("EVENT", contains("K40W"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D7K40W2_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K40W3_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K40W4_191011_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
K40w_events <- K40w$EVENT

K60w <- inc %>% select(c("EVENT", contains("K60W"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D7K60W2_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K60W3_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K60W4_190919_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
K60w_events <- K60w$EVENT

L20d <- inc %>% select(c("EVENT", contains("L20d"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D6L20d1_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6L20d2_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6L20d3_190919_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
L20d_events <- L20d$EVENT

L40d <- inc %>% select(c("EVENT", contains("L40d"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D6L40d1_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L40d2_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L40d3_190919_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
L40d_events <- L40d$EVENT

L60d <- inc %>% select(c("EVENT", contains("L60d"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D6L60d1_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L60d2_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L60d3_190903_NextSeq_R1-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
L60d_events <- L60d$EVENT

L20w <- inc %>% select(c("EVENT", contains("L20W"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D6L20W1_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L20W2_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L20W3_190903_NextSeq_R1-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
L20w_events <- L20w$EVENT

L40w <- inc %>% select(c("EVENT", contains("L40W"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D6L40W1_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L40W2_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L40W3_190903_NextSeq_R1-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
L40w_events <- L40w$EVENT

L60w <- inc %>% select(c("EVENT", contains("L60W"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D6L60W1_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L60W2_190903_NextSeq_R1-Q`, 1, 1) == "N" | str_sub(`D6L60W3_190903_NextSeq_R1-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
L60w_events <- L60w$EVENT

Krw <- inc %>% select(c("EVENT", contains("Krw"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D6Krw1_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6Krw3_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7Krw2_190919_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
Krw_events <- Krw$EVENT

Lrw <- inc %>% select(c("EVENT", contains("Lrw"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D6Lrw1_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6Lrw2_190919_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6Lrw3_190919_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  filter(AS_percent >= 25)
Lrw_events <- Lrw$EVENT
```

change this to all the individual lists
```{r}
all_events <- as.data.frame(unique(c(K20d_events, K40d_events, K60d_events, K20w_events, K40w_events, K60w_events, Krw_events, L20d_events, L40d_events, L60d_events, L20w_events, L40w_events, L60w_events, Lrw_events)))
colnames(all_events) <- "event"
```

then
```{r}
db <- read_delim("C:/Users/hasna/Documents/MSc_project/vasttools/vast-tools/Ath/FILES/Ath.Event-Gene.IDs.txt", delim = "\t")
colnames(db) <- c("EVENT", "GeneID")

all_events <- all_events %>% mutate(type = case_when(
  grepl("AthEX.*", event) ~ "Exon skipping",
  grepl("AthALTA.*", event) ~ "Alternative 3' SS",
  grepl("AthALTD", event) ~ "Alternative 5' SS",
  grepl("AthINT", event) ~ "Intron retention"
))
ES <- all_events %>% filter(type == "Exon skipping") %>% 
  select(event)
colnames(ES) <- "EVENT"
ALTA <- all_events %>% filter(type == "Alternative 3' SS") %>% 
  select(event)
colnames(ALTA) <- "EVENT"
ALTD <- all_events %>% filter(type == "Alternative 5' SS") %>% 
  select(event)
colnames(ALTD) <- "EVENT"
IR <- all_events %>% filter(type == "Intron retention") %>% 
  select(event)
colnames(IR) <- "EVENT"
ES <- merge(ES, db, by = "EVENT", all.x = T)
ES_genes <- unique(ES$GeneID)
ALTA <- merge(ALTA, db, by = "EVENT", all.x = T)
ALTA_genes <- unique(ALTA$GeneID)
ALTD <- merge(ALTD, db, by = "EVENT", all.x = T)
ALTD_genes <- unique(ALTD$GeneID)
IR <- merge(IR, db, by = "EVENT", all.x = T)
IR_genes <- unique(IR$GeneID)

to_plot_2 <- list(ES_genes, IR_genes, ALTA_genes, ALTD_genes)
```

plot AS events by type
```{r}
tiff("test.tiff", units="in", width=20, height=15, res=300)
ggVennDiagram(to_plot_2,
              label_color = "white",
              label_alpha = 0,
              category.names = c(paste("Exon", "\n","skipping", sep = ""), "Intron retention", "Alternative 3' SS", paste("Alternative", "\n", "5' SS", sep = "")),
              label_size = 10,
              set_size = 10,
              edge_size = 2)+
  scale_fill_gradient(high = "green", low = "grey")+
  theme(legend.key.size = unit(2, "cm"),
        legend.title = element_blank(),
        legend.text = element_text(size = 15))
dev.off()
```

bar graph
```{r}
K20d$name <- c("K20d")
K40d$name <- c("K40d")
K60d$name <- c("K60d")
K20w$name <- c("K20w")
K40w$name <- c("K40w")
K60w$name <- c("K60w")
Krw$name <- c("Krw")
L20d$name <- c("L20d")
L40d$name <- c("L40d")
L60d$name <- c("L60d")
L20w$name <- c("L20w")
L40w$name <- c("L40w")
L60w$name <- c("L60w")
Lrw$name <- c("Lrw")

K20d_short <- select(K20d, name, EVENT)
K40d_short <- select(K40d, name, EVENT)
K60d_short <- select(K60d, name, EVENT)
K20w_short <- select(K20w, name, EVENT)
K40w_short <- select(K40w, name, EVENT)
K60w_short <- select(K60w, name, EVENT)
Krw_short <- select(Krw, name, EVENT)
L20d_short <- select(L20d, name, EVENT)
L40d_short <- select(L40d, name, EVENT)
L60d_short <- select(L60d, name, EVENT)
L20w_short <- select(L20w, name, EVENT)
L40w_short <- select(L40w, name, EVENT)
L60w_short <- select(L60w, name, EVENT)
Lrw_short <- select(Lrw, name, EVENT)

bgdata <- rbind(K20d_short, K40d_short, K60d_short, K20w_short, K40w_short, K60w_short, Krw_short, L20d_short, L40d_short, L60d_short, L20w_short, L40w_short, L60w_short, Lrw_short)

bgdata <- bgdata %>% mutate(type = case_when(
  grepl("AthEX.*", EVENT) ~ "Exon skipping",
  grepl("AthALTA.*", EVENT) ~ "Alternative 3' SS",
  grepl("AthALTD", EVENT) ~ "Alternative 5' SS",
  grepl("AthINT", EVENT) ~ "Intron retention"
)) %>% 
  group_by(name, type) %>% 
  summarize(count = n())

colours = c("grey70", "chartreuse2", "palevioletred1", "cadetblue2")
```

```{r}
ggplot(bgdata, aes(x = name, y = count, fill = type))+
  geom_bar(stat = "identity")+
  scale_fill_manual(values = colours)+
  theme_minimal()
```
bar graph differential splicing
```{r}
K20dK20w <- read_delim("dPSI25_no_merging_all_reps/K20d-K20w_68events/K20d-K20w_compare_ouput_complete.tab") %>% 
  select(Event_type) %>% 
  group_by(Event_type) %>% 
  summarize(count = n())
K20dK20w$name = c("K20dK20w")

K40dK40w <- read_delim("dPSI25_no_merging_all_reps/K40d-K40w_23events/K40d-K40w_compare_ouput_complete.tab") %>% 
  select(Event_type) %>% 
  group_by(Event_type) %>% 
  summarize(count = n())
K40dK40w$name = c("K40dK40w")

K60dK60w <- read_delim("dPSI25_no_merging_all_reps/K60d-K60w_6events/K60d-K60w_compare_ouput_complete.tab") %>% 
  select(Event_type) %>% 
  group_by(Event_type) %>% 
  summarize(count = n())
K60dK60w$name = c("K60dK60w")

allcomps <- rbind(K20dK20w, K40dK40w, K60dK60w)
```

```{r}
ggplot(allcomps, aes(x = name, y = count, fill = Event_type))+
  geom_bar(stat = "identity")+
  scale_fill_manual(values = colours)+
  theme_minimal()
```

May 2022: volcano plot. need to read in the inclusion table, filter for appropriate read depth, average the PSIs, then calculate dPSI for the K20dK20w comparison. This is close to recapitulating vast compare, except we don't require a range difference of 5 PSI between the 2 groups. It will be interesting to see what the effect of this cutoff, and of requiring support from majiq, are on the datapoints we chose out of all the valid vast events (this will be the difference between points with |dPSI| > 25, but that we're selected. Do these have lower p-values as well?)


```{r}
orig_inc <- inc <- read_delim("C:/Users/hasna/Documents/MSc_project/vasttools/all_data/complete_inclusion_table.tab", delim = "\t") 

K20d <- inc %>% select(c("EVENT", contains("K20d"))) %>% 
  mutate("pass" = ifelse(
  str_sub(`D6K20d2_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6K20d3_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K20d2_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K20d5_191011_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent_K20d" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  select(EVENT, AS_percent_K20d) 

K20w <- inc %>% select(c("EVENT", contains("K20W"))) %>% 
  mutate("pass" = ifelse(
str_sub(`D7K20W1_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D7K20W2_191011_NextSeq-Q`, 1, 1) == "N" | str_sub(`D6K20W1_191011_NextSeq-Q`, 1, 1) == "N", "no", "yes")) %>% 
  filter(pass == "yes") %>% 
  mutate("mean" = rowMeans(.[, unlist(lapply(., is.numeric))])) %>% 
  mutate("AS_percent_K20w" = ifelse(
  grepl("AthEX.*", EVENT), 100 - mean, mean
  )) %>% 
  select(EVENT, AS_percent_K20w)

K20dK20w <- inner_join(K20d, K20w, by = "EVENT") %>% 
  mutate(dPSI = AS_percent_K20d - AS_percent_K20w)
test <- filter(K20dK20w, dPSI < -25) #more events that we'd expect since the 5 PSI range filter is not in effect. all we're looking for is valid splicing events (not dAS)

#add gene names
db <- read_delim("C:/Users/hasna/Documents/MSc_project/vasttools/vast-tools/Ath/FILES/Ath.Event-Gene.IDs.txt", delim = "\t")
colnames(db) <- c("EVENT", "GeneID") #gotta get the colnames to match across DFs
K20dK20w <- merge(K20dK20w, db, by = "EVENT", all.x = T)
```

join K20dK20w (contains dPSI) to diff (contains MV[dPsi]_at_0.95). we can also interpret this as the lower bound of the 90% CI of dPSI. get the dAS events we identified in K20dK20w
```{r}
diff <- read_delim("C:/Users/hasna/Documents/MSc_project/vasttools/vast-tools/vast_out/diff_out_feb2022/diff_out95.tab", delim = "\t")
K20dK20w <- inner_join(K20dK20w, diff, by = "EVENT")

vm_K20dK20w <- read_delim("C:/Users/hasna/Documents/MSc_project/majiq/dAS_analysis/clustering/dAS_DEG/all_events.tsv", delim = "\t") %>% 
  filter(comp == "K20dK20w", set == "dAS", event != "AthINT0088627")
#get rid of the second event in AT5G49940 that majiq didn't detect

to_label <- filter(K20dK20w, EVENT %in% vm_K20dK20w$event)
```

volcano plot!
```{r}
ggplot(K20dK20w, aes(x = dPSI, y = `MV[dPsi]_at_0.95`, label = GeneID))+
  geom_point(colour = "red")+
  gghighlight(EVENT %in% vm_K20dK20w$event, unhighlighted_params = list(colour = "grey"))+
  geom_vline(xintercept = 25, linetype = 2)+
  geom_vline(xintercept = -25, linetype = 2)+
  geom_text_repel(data = to_label, aes(label = GeneID), size = 2.5)
```

